### PR TITLE
Feat/temporal rasterisation

### PR DESCRIPTION
**Pull request by batic**
Load (and save) eopatch timestamps if rasterisation for temporal features is requested.

### CODE REVIEW

File: eogrow/pipelines/rasterize.py [hunk 1]
```diff
@@ -252,5 +255,9 @@ def _get_dataset_path(self, filename: str) -> str:
     def _get_output_features(self) -> List[FeatureSpec]:
         """Lists all features that are to be saved upon the pipeline completion"""
         features: List[FeatureSpec] = [FeatureType.BBOX]
+
+        if self.vector_feature[0].is_temporal():
```
Comment (mlubej) [inline eogrow/pipelines/rasterize.py:259]: As you mentioned, you're bothered by the `self.vector_feature[0].is_temporal()`, which I agree with. The only thing I can come up with is defining the following

```python
f_type, _ = self.vector_feature
```

and then using `f_type.is_temporal()`. No need to take this into account though. Approving, but I'm letting you decide.

**Review by mlubej — APPROVED** [review]

File: eogrow/pipelines/rasterize.py (hunk 1 above)
Comment (zigaLuksic) [reply eogrow/pipelines/rasterize.py:259]: another option is to have a static method `vector_feature_is_temporal(feature)` and hide that in there.
Sure it's a one-liner, but if it makes the code easier to read...

File: eogrow/pipelines/rasterize.py [hunk 2]
```diff
@@ -252,5 +255,9 @@ def _get_dataset_path(self, filename: str) -> str:
...
+            features.extend([FeatureType.TIMESTAMP])
```
Comment (zigaLuksic) [inline eogrow/pipelines/rasterize.py:260]: any reason for extend instead of `append`?

Comment (batic) [reply eogrow/pipelines/rasterize.py:260]: none whatsoever

File: eogrow/pipelines/rasterize.py (hunk 1 above)
Comment (batic) [reply eogrow/pipelines/rasterize.py:259]: can do oneliner static method, but wasn't sure where to add it; should it be in the class, or in utils, or ....

Comment (zigaLuksic) [reply eogrow/pipelines/rasterize.py:259]: I would either add it as part of the class, or add it as a utility function at the end of the file with a `_` prefix so it's not in the docs.

File: eogrow/pipelines/rasterize.py [hunk 3]
```diff
@@ -103,12 +103,12 @@ def __init__(self, *args: Any, **kwargs: Any):
...
         self.filename: Optional[str] = None
 
-        if len({c.output_feature[0].is_temporal() for c in self.config.columns}) != 1:
+        if len({self._is_temporal(c.output_feature) for c in self.config.columns}) != 1:
             raise ValueError("All output features have to be either temporal or timeless!")
 
         if isinstance(self.config.vector_input, str):
             self.filename = self._parse_input_file(self.config.vector_input)
-            output_is_temporal = self.config.columns[0].output_feature[0].is_temporal()
+            output_is_temporal = self._is_temporal(self.config.columns[0].output_feature)
```
Comment (zigaLuksic) [inline eogrow/pipelines/rasterize.py:111]: shouldn't this check ALL output features?

**Review by zigaLuksic — APPROVED** [review]

File: eogrow/pipelines/rasterize.py (hunk 3 above)
Comment (batic) [reply eogrow/pipelines/rasterize.py:111]: I agree.

It is partially already done; there is a check for input features to be all temporal or timeless. So the final (missing) check is that output feature also conforms to input feature timeliness.

Comment (batic) [reply eogrow/pipelines/rasterize.py:111]: Added the (quick) test, we can address how to check this properly (at schema validation phase?) in another issue.

**Comment by zigaLuksic** [comment]
@mlubej any other suggestions?

**Comment by mlubej** [comment]
> @mlubej any other suggestions?

No, good to go AFAIC.

**Pull request merged**